### PR TITLE
Richtext product page fields

### LINKED ISSUE #7694: [PNI CMS QA] Update multiple "plain text" fields to "rich text"
More fields now need to allow for url links. Please make the follow fields richtext:

- [x] Know track record description
- [x] How does the company use this data?
- [x] How can you control your data?
- [x] General AI description
- [x] Manages security vulnerabilities Description

Reminder to also do what will be done in https://github.com/mozilla/foundation.mozilla.org/issues/7692

### PR DESCRIPTION
**Pull request by danielfmiranda**
Closes #7694 
Related PRs/issues #

**Link to sample test page**: https://foundation-s-richtext-p-hhzunw.mofostaging.net/en/privacynotincluded/between-would-deal-effort-table-ability-technology-evidence/

**Steps to test:**

1. Please visit the test page linked above
2. Scroll down to the following fields, and check if they are rich text:
- [ ]  What is the company’s known track record of protecting users’ data? (Privacy Tab)
- [ ] How does the company use this data? (Privacy Tab)
- [ ] How can you control your data? (Privacy Tab)
- [ ] General AI description/helptext (AI tab)
- [ ] Manages security vulnerabilities Description/helptext (security tab)
3. If everything is looking/behaving correctly, testing is complete! 

Please note that since Faker and Factory run after migrations, the rest of the pages are not going to render the text styled as the test page.
When this gets merged into staging/prod, the new migration will take all existing pages and update them as expected.

### Screenshots of fields on review app:
image

image

image

## Checklist
**Changes in Models:**
- [x] [Are my changes backward-compatible](https://github.com/mozilla/foundation.mozilla.org/blob/main/docs/workflow.md#django-migrations-what-to-do-when-working-on-backward-incompatible-migrations). If not, did I schedule a deploy with the rest of the team

### CODE REVIEW

File: network-api/networkapi/wagtailpages/factory/buyersguide.py
Comment (danielfmiranda) [inline network-api/networkapi/wagtailpages/factory/buyersguide.py:225]: I updated these two fields so Percy would not mark them as unstyled, since the CSS rules are now looking for a <p> element

File: network-api/networkapi/wagtailpages/migrations/0049_pni_criterion_richtext_fields.py
Comment (danielfmiranda) [inline network-api/networkapi/wagtailpages/migrations/0049_pni_criterion_richtext_fields.py:51]: Update requested by pomax, we are now getting the default locale using this function, then using the ID to query for product pages.

File: network-api/networkapi/wagtailpages/templates/buyersguide/product_page.html
Comment (danielfmiranda) [inline network-api/networkapi/wagtailpages/templates/buyersguide/product_page.html:279]: Line 280 seemed to be there by mistake, i moved it up to line 262

File: network-api/networkapi/wagtailpages/pagemodels/products.py
Comment (danielfmiranda) [inline network-api/networkapi/wagtailpages/pagemodels/products.py:1173]: Upon request of Pomax, we have created a ticket to unify these choices and have them in a new list, which we can call whenever making/updating a rich text field.

https://github.com/mozilla/foundation.mozilla.org/issues/7755

File: network-api/networkapi/wagtailpages/templates/buyersguide/product_page.html
Comment (danielfmiranda) [inline network-api/networkapi/wagtailpages/templates/buyersguide/product_page.html:216]: Per request, I created a new ticket to update this template to have it sense whether or not it is using a rich text value, instead of having to set it through a boolean each time. 

https://github.com/mozilla/foundation.mozilla.org/issues/7755

File: network-api/networkapi/wagtailpages/templates/fragments/product_criterion_primary_info.html
Comment (danielfmiranda) [inline network-api/networkapi/wagtailpages/templates/fragments/product_criterion_primary_info.html:18]: Upon further investigation, I noted that this template needs some updating. I have created a ticket with further details here: https://github.com/mozilla/foundation.mozilla.org/issues/7757

**Comment by danielfmiranda** [comment]
Hi @Pomax!

I have implemented your feedback and also created the tickets that you have requested in the previous review. 

Thank you in advance for reviewing this ticket!

**Review by kristinashu — APPROVED** [review]
The design looks good! Thank you for setting up such a detailed test page.

**Review by Pomax — CHANGES_REQUESTED** [review]
looks good, but this will need the migrations updated before final approval

**Review by Pomax — APPROVED** [review]

**Pull request merged**
